### PR TITLE
[FIX] event_sale: flush out state field

### PR DESCRIPTION
**Pull request by sswapnesh**
Description of the issue/feature this PR addresses: Fixes https://github.com/odoo/odoo/issues/38340 and Followup on https://github.com/odoo/odoo/commit/9920f20e4c7753bc17bea71dea3a90f7de687196#diff-d9c980d61cddfd4f76333e7a5b92552bR204

Current behavior before PR: See https://github.com/odoo/odoo/issues/38340

Desired behavior after PR is merged: Correct Recomputation

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by sswapnesh** [comment]
I am not sure about the Diff and commit msg but this solution seems to be working.

Cc @rco-odoo @mart-e

File: addons/event_sale/models/event.py [hunk 1]
```diff
@@ -133,6 +133,7 @@ def _compute_seats(self):
                         WHERE event_ticket_id IN %s AND state IN ('draft', 'open', 'done')
                         GROUP BY event_ticket_id, state
                     """
+            self.env['event.registration'].flush(['event_id', 'state'])
```
Comment (rco-odoo) [inline addons/event_sale/models/event.py]: Why don't you flush `event_ticket_id`, too?
```suggestion
            self.env['event.registration'].flush(['event_id', 'event_ticket_id', 'state'])
```

Comment (sswapnesh) [reply addons/event_sale/models/event.py]: Updated

**Comment by mart-e** [comment]
@sswapnesh in your commit message, please try to explain the _why_.
Only referring to a commit is usually not very clear (or not enough).
You specify an issue in the description of the PR but not the commit. Would be worth adding it there too.
Thanks

**Comment by sswapnesh** [comment]
:thinking:  
**[FIX] event_sale: flush out fields to recompute state correctly**
Followup on 9920f20#diff-d9c980d61cddfd4f76333e7a5b92552bR204



Will it be okay?

**Comment by mart-e** [comment]
It is always good to explain the consequences of the problem. "recompute X correctly" says _what_ the commit does but the important part is _why_ do we need to recompute it. Something like "before this commit Y was wrong as X was not recomputed".
Let say, in two years, we notice a side-effect linked to this commit, we need to understand, reading only the commit message. if the root cause of the patch still applies today (because nobody will remember the bug by them).

Also, please add the reference to the issue in the commit message too (i.e. `Fixes odoo/odoo#38340`). Even if github issues are not as durable as commit message, it is still a good reference information (and it will [automatically close the issue](https://help.github.com/en/github/managing-your-work-on-github/closing-issues-using-keywords)).

Now that you are a recurrent (and awarded!) contributor, we ask you a bit more work on the PRs :)
Thanks

**Comment by mart-e** [comment]
> Before this commit 'state' was not commputed correctly due to API changes made at https://github.com/odoo/odoo/commit/9920f20e4c7753bc17bea71dea3a90f7de687196 for stored Compute fields.

You are still not explaining what the bug is. It is important that, when reading _only the commit message_, you understand the bug that you are fixing. Here you are only explaining the cause of the bug (api change), you are not explaining why this is a problem (the bug described at #38340).
Saying "state is not corrected computed due to 9920f20e4c7753" does not explain the problem (the SQL query result may be desynchronised with potential changes made by the ORM but not propagated yet to the database).
Why do I need to make a `flush` and not another operation?

This gives you the bigger picture and, as explained above, is good information if we realise later the patch is incorrect.

Read again the [git commit guidelines](https://www.odoo.com/documentation/12.0/reference/guidelines.html#git), we give examples of good messages and how it should be structured formally.
If you are not sure, try to explain how to reproduce the bug.
See for instance 272359b86c030c814a the diff is very small but it really explains what the problem was, all commit messages should look like that (I know it is not easy).

Anyway, I have modified the commit message (please read it) so we can merge now :)
@robodoo r+

**Comment by sswapnesh** [comment]
This will be very tuff job for me🙇‍♂️

**Pull request closed without merging**